### PR TITLE
Makefile improvements

### PR DESCRIPTION
**Pull request by clebergnu**
This is a small collection of Makefile related cleanups and improvements.

### CODE REVIEW

File: Makefile [hunk 1]
```diff
@@ -184,7 +184,19 @@ spell:
 
 man: man/avocado.1 man/avocado-rest-client.1
 
-.PHONY: source install clean check link
+show:
```
Comment (ldoktor) [inline Makefile]: Hmm, show what? I think better would be `make environment`, `make variables` or `make debug`.

Comment (ldoktor) [inline Makefile]: also worth mentioning I like the patch a lot :smile:

File: Makefile [hunk 2]
```diff
@@ -83,25 +75,6 @@ pypi: source-pypi develop
...
-	dpkg-buildpackage -S -elookkas@gmail.com -rfakeroot
-
-deb-bin: deb-prepare-source
-	# build binary package
-	dpkg-buildpackage -b -rfakeroot
-
-deb: deb-prepare-source
-	# build both source and binary packages
-	dpkg-buildpackage -i -I -rfakeroot
-
```
Comment (ldoktor) [inline Makefile]: Well it'd be nice to include the basic workflow in the contrib script. Can you please add it? (nothing fancy, just README or INSTALL file in the `debian` directory.

Comment (clebergnu) [inline Makefile]: Actually I forgot to add the `contrib/packaging/debian/Makefile`.  Sending that in v2.

**Comment by clebergnu** [comment]
Closing in favor of #1431

**Pull request closed without merging**